shapely/geometry/point.py

```diff
@@ -64,12 +64,17 @@ def __new__(self, *args):
             # Accept either (x, y) or [(x, y)]
             if not hasattr(coords, "__getitem__"):  # generators
                 coords = list(coords)
-
             coords = np.asarray(coords).squeeze()
         else:
             # 2 or 3 args
             coords = np.array(args).squeeze()
 
+        if coords.ndim > 1:
+            raise ValueError(
+                f"Point() takes only scalar or 1-size vector arguments, got {args}"
+            )
+        if not np.issubdtype(coords.dtype, np.number):
+            coords = [float(c) for c in coords]
         geom = shapely.points(coords)
         if not isinstance(geom, Point):
             raise ValueError("Invalid values passed to Point constructor")
```

Comment:
@idanmiara I added this back. I know this is currently not needed (I don't think a test covers this), and I also can't directly think of a case where `shapely.points(..)` would not return a single Point that wouldn't already be catched by the `if coords.ndim > 1:` check. 
But, Point() really should ever only return a single Point, and `shapely.points()` doesn't guarantee that, so it feels safer to keep this anyway (our tests might not be complete).